core/renderers/block_rendering_rewrite/block_render_draw.js

```diff
@@ -91,7 +103,9 @@ Blockly.blockRendering.Drawer.prototype.drawOutline_ = function() {
   this.drawTop_();
   for (var r = 1; r < this.info_.rows.length - 1; r++) {
     var row = this.info_.rows[r];
-    if (row.hasStatement) {
+    if (!row.isSpacer() && this.info_.isCollapsed) {
```

Comment:
Instead of checking `this.info_.isCollapsed`, add a `hasJaggedEdge` property to the row and check that property.

Reply:
Done